overlord/snapstate/storehelpers.go

```diff
@@ -693,17 +726,38 @@ func storeUpdatePlan(
 			return updatePlan{}, err
 		}
 
-		// TODO:COMPS: handle components here
+		// TODO:COMPS: for now, go back to the components that were already
+		// installed with this revision. to be more robust, we'd need to compare
+		// what components are installed with the revision that we are
+		// refreshing from to the components available in the store for the
+		// revision that we are refreshing to.
```

Comment:
I'll reword to make things more clear. Although, we could look at the components in the older sequence and compare to what the store reports if we want to save a download.

Reply:
Wait, just to be clear, this comment is related to what we should do when we `snap refresh --revision=n`, where `n` has already been installed.

Reply:
But would this be for `n` being the snap currently installed or for `n` being a revision that was installed in the past and because of that we have it locally? (but we are not really reverting to that sequence point in that case).

Reply:
Correct, this would be for a revision `n` that has been previously installed, but we are not reverting to it. We are refreshing to it again.

Reply:
Then I think that my original comment applies, we do not want to look at the components that were installed in the past for that rev, but to the ones we have now and refresh those only (if they exist).

Reply:
do we need tests or TODO:COMPS about Revert specifically ?